sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,211 @@
+  {}
+
+private:
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_;
+};
+
+void BenchmarkShouldSampler(Sampler &sampler, benchmark::State &state)
+{
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(sampler.ShouldSample(nullptr, trace_id, "", span_kind, view));
+	}
+}
+
+namespace
+{
+
+// Sampler constructor used as a baseline to compare with other samplers
+void BM_AlwaysOffSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOffSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOffSamplerConstruction);
+
+// Sampler constructor used as a baseline to compare with other samplers
+void BM_AlwaysOnSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOnSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOnSamplerConstruction);
+
+void BM_ParentOrElseSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ParentOrElseSampler(std::make_shared<AlwaysOnSampler>()));
+	}
+}
+BENCHMARK(BM_ParentOrElseSamplerConstruction);
+
+void BM_ProbabilitySamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ProbabilitySampler(0.01));
+	}
+}
+BENCHMARK(BM_ProbabilitySamplerConstruction);
+
+// Sampler used as a baseline to compare with other samplers
+void BM_AlwaysOffSamplerShouldSample(benchmark::State &state)
+{
+	AlwaysOffSampler sampler;
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_AlwaysOffSamplerShouldSample);
+
+// Sampler used as a baseline to compare with other samplers
+void BM_AlwaysOnSamplerShouldSample(benchmark::State &state)
+{
+	AlwaysOnSampler sampler;
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_AlwaysOnSamplerShouldSample);
+
+void BM_ParentOrElseSamplerShouldSample(benchmark::State &state)
+{
+	ParentOrElseSampler sampler(std::make_shared<AlwaysOnSampler>());
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_ParentOrElseSamplerShouldSample);
+
+void BM_ProbabilitySamplerShouldSample(benchmark::State &state)
+{
+	ProbabilitySampler sampler(0.01);
+
+	BenchmarkShouldSampler(sampler, state);
+}
+BENCHMARK(BM_ProbabilitySamplerShouldSample);
+
+// Test to measure performance for span creation
+void BM_SpanCreation(benchmark::State &state)
+{
+	std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_off(
+		new std::vector<std::unique_ptr<SpanData>>);
+
+	std::unique_ptr<SpanExporter> exporter(new MockSpanExporter(spans_received_off));
+	auto processor = std::make_shared<SimpleSpanProcessor>(std::move(exporter));
```

Comment:
You should be able to directly pass "std::make_unique<MockSpanExporter>(...)" as a param. Then there's no need for "std::move" since C++ will elide copy/move operations in this case. See https://en.cppreference.com/w/cpp/language/copy_elision.

Reply:
Thanks - fixed